clusterloader2/GETTING_STARTED.md

```diff
@@ -0,0 +1,227 @@
+name: test
+# Number of namespaces that will be created
+namespace:
+  number: 1
+# Tuning set with qps 1 will execute 1 action every second.
+# In out case, creating deployment will be 1 action to execute.
+tuningSets:
+- name: Uniform1qps
+  qpsLoad:
+    qps: 1
+# Test consists of list of steps. A step can be either collection of Phases or Measurements.
+# A Phase defines a state the cluster should reach.
+# A Measurement allows to measure something or wait for something.
+# You can find list of available measurements here: https://github.com/kubernetes/perf-tests/tree/master/clusterloader2#measurement
+steps:
+# We want to start two measurement.
+# One will measure time it takes to start a pod, second will make sure we wait for pod to be in running state.
+# Setting the action field to "start" begins execution of measurement.
+- name: Start measurements
+  measurements:
+  - Identifier: PodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: start
+# This label specifies for which pods we want to measure pod startup latency.
+      labelSelector: group = test-pod
+# If pod startup latency goes over threshold, test will fail.
+      threshold: 5s
+  - Identifier: WaitForControlledPodsRunning
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: start
+      apiVersion: apps/v1
+# Type of object we are waiting for
+      kind: Deployment
+      labelSelector: group = test-pod
+# How long we want to wait till object is in running state.
+      operationTimeout: 120s
+# Once we created these two measurements, we can create deployment in next step.
+- name: Create deployment
+  phases:
+# namespaceRange specifies in which namespaces we want to create this deployment.
+  - namespaceRange:
+      min: 1
+      max: 1
+# Here we an specify how many replicas of object we want to create within namespace.
+    replicasPerNamespace: 1
+    tuningSet: Uniform1qps
+    objectBundle:
+    - basename: test-deployment
+# Path to file containing template for object to be created.
+      objectTemplatePath: "deployment.yaml"
+# Parameters to fill template.
+      templateFillMap:
+        Replicas: 1
+# Once we started measurements and created deployment, now we can wait for this deployment to be running.
+# Also, we can measure pod startup latency
+- name: Wait for pods to be running
+  measurements:
+  - Identifier: WaitForControlledPodsRunning
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: gather
+  - Identifier: PodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: gather
+```
+Now, we need to prepare template for our deployment to be created. All values in brackets will be filled from template parameters:
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+# Name is parameter that will be autogenerated by ClusterLoader
+  name: {{.Name}}
+  labels:
+    group: test-pod
+spec:
+  replicas: {{.Replicas}}
+  selector:
+    matchLabels:
+      name: {{.Name}}
```

Comment:
Can we go step further and simplify further?
What we really use in test is group: test-pod, so let's add group:test-pod here and remove "name: <name>" labels/selectors from everywhere

Reply:
Done.